internal/service/kubernetes/cluster_data_source.go

```diff
@@ -0,0 +1,161 @@
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"gopkg.in/yaml.v3"
+)
+
+func NewKubernetesClusterDataSource() datasource.DataSource {
+	return &kubernetesClusterDataSource{}
+}
+
+var (
+	_ datasource.DataSource              = &kubernetesClusterDataSource{}
+	_ datasource.DataSourceWithConfigure = &kubernetesClusterDataSource{}
+)
+
+type kubernetesClusterDataSource struct {
+	client *service.Service
+}
+
+func (d *kubernetesClusterDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_kubernetes_cluster"
+}
+
+func (d *kubernetesClusterDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	d.client, resp.Diagnostics = utils.GetClientFromProviderData(req.ProviderData)
+}
+
+type kubernetesClusterDataModel struct {
+	ClientCertificate    types.String `tfsdk:"client_certificate"`
+	ClientKey            types.String `tfsdk:"client_key"`
+	ClusterCACertificate types.String `tfsdk:"cluster_ca_certificate"`
+	ID                   types.String `tfsdk:"id"`
+	Host                 types.String `tfsdk:"host"`
+	Kubeconfig           types.String `tfsdk:"kubeconfig"`
+	Name                 types.String `tfsdk:"name"`
+}
+
+func (m *kubernetesClusterDataModel) setB64Decoded(field string, encoded string) (diags diag.Diagnostics) {
+	decoded, err := base64.StdEncoding.DecodeString(encoded)
+	if err != nil {
+		diags.AddError(
+			fmt.Sprintf("Unable to decode base64 encoded value of %s", field),
+			utils.ErrorDiagnosticDetail(err),
+		)
+		return
+	}
+
+	value := types.StringValue(string(decoded))
+	reflect.ValueOf(m).Elem().FieldByName(field).Set(reflect.ValueOf(value))
+	return
+}
+
+func (d *kubernetesClusterDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Managed Kubernetes cluster details. Please refer to [Terraform documentation on sensitive data](https://www.terraform.io/language/state/sensitive-data) to keep the credential data as safe as possible.",
+		Attributes: map[string]schema.Attribute{
+			"client_certificate": schema.StringAttribute{
+				Description: clientCertificateDescription,
+				Computed:    true,
+			},
+			"client_key": schema.StringAttribute{
+				Description: clientKeyDescription,
+				Computed:    true,
+				Sensitive:   true,
+			},
+			"cluster_ca_certificate": schema.StringAttribute{
+				Description: clusterCACertificateDescription,
+				Computed:    true,
+			},
+			"id": schema.StringAttribute{
+				Required:    true,
+				Description: idDescription,
+			},
+			"host": schema.StringAttribute{
+				Computed:    true,
+				Description: hostDescription,
+			},
+			"kubeconfig": schema.StringAttribute{
```

Comment:
Should `kubeconfig` also be sensitive as it includes the `client_key` value? 🤔

Reply:
Makes sense to mark it as sensitive